pulp_smash/tests/rpm/cli/test_upload.py

```diff
@@ -0,0 +1,119 @@
+# coding=utf-8
+"""Tests upload of units via client."""
+from __future__ import unicode_literals
+
+import inspect
+import os
+
+import unittest2
+
+from pulp_smash import cli, config, utils
+from pulp_smash.constants import DRPM_URL, DRPM
+from pulp_smash.tests.rpm.utils import set_up_module
+
+
+def setUpModule():  # pylint:disable=invalid-name
+    """Execute ``pulp-admin login`` on the target Pulp system."""
+    set_up_module()
+    utils.pulp_admin_login(config.get_config())
+
+
+class UploadBaseTestCase(unittest2.TestCase):
```

Comment:
Is an abstract base class needed, given that there's exactly one child of this class? Let's add complexity when it's needed. Doing so makes the code easier to understand and maintain in its current state, and it makes the code easier to refactor in the future.